main/src/cgeo/geocaching/filter/CountryFilterFactory.java

```diff
@@ -67,23 +68,31 @@ public boolean accepts(@NonNull final Geocache cache) {
     @NonNull
     public List<IFilter> getFilters() {
 
-        final Map<String, IFilter> filters = new HashMap<>();
-
+        // which countries do we have?
+        final Map<String, String> tempUnsorted = new HashMap<>();
```

Comment:
just use a LinkedHashSet, then we don't need to sort again

Reply:
LinkedHashMap is a good hint, thanks for that, but does not solve the problem completely, see comment below.